ts/features/fci/components/DocumentViewer.tsx

```diff
@@ -0,0 +1,221 @@
+            url
+          )
+            .then(_ => {
+              showToast(
+                I18n.t(
+                  "features.mvl.details.attachments.pdfPreview.savedAtLocation",
+                  {
+                    name: "attachment.displayName"
+                  }
+                ),
+                "success"
+              );
+            })
+            .catch(_ => {
+              showToast(
+                I18n.t(
+                  "features.mvl.details.attachments.pdfPreview.errors.saving"
+                )
+              );
+            });
+        },
+        title: I18n.t("features.mvl.details.attachments.pdfPreview.save")
+      }}
+      rightButton={confirmButtonProps(() => {
+        onOpen?.();
+        ReactNativeBlobUtil.android
+          .actionViewIntent(url, "application/pdf")
+          .catch(_ => {
+            showToast(
+              I18n.t(
+                "features.mvl.details.attachments.pdfPreview.errors.opening"
+              )
+            );
+          });
+      }, I18n.t("features.mvl.details.attachments.pdfPreview.open"))}
+    />
+  );
+
+type Props = {
+  documentUrl: string;
+  onLoadComplete?: () => void;
+  onError?: () => void;
+  onShare?: () => void;
+  onOpen?: () => void;
+  onDownload?: () => void;
+};
+
+export const DocumentViewer = (props: Props): React.ReactElement => {
+  const [isError, setIsError] = useState(false);
+  const documentUrl = props.documentUrl;
+  const dispatch = useIODispatch();
+  const navigation = useNavigation();
+  const fciDownloadSelector = useIOSelector(fciDownloadPreviewSelector);
+  const fciDownloadPath = useIOSelector(fciDownloadPathSelector);
+
+  React.useEffect(() => {
+    dispatch(fciDownloadPreview.request({ url: documentUrl }));
+  }, [documentUrl, dispatch]);
+
+  const LoadingComponent = () => (
+    <LoadingErrorComponent
+      isLoading={true}
+      loadingCaption={""}
+      onRetry={constNull}
+      testID={"FciRouterLoadingScreenTestID"}
+    />
+  );
+
+  if (pot.isLoading(fciDownloadSelector)) {
+    return <LoadingComponent />;
+  }
+
+  const customGoBack: React.ReactElement = (
+    <TouchableDefaultOpacity
+      onPress={() => {
+        dispatch(fciDownloadPreviewCancel({ path: fciDownloadPath }));
+        navigation.goBack();
+      }}
+      accessible={true}
+      accessibilityLabel={I18n.t("global.buttons.back")}
+      accessibilityRole={"button"}
+    >
+      <IconFont name={"io-back"} style={{ color: IOColors.bluegrey }} />
+    </TouchableDefaultOpacity>
+  );
+
+  return !pot.isError(fciDownloadSelector) ? (
+    <BaseScreenComponent
+      goBack={true}
+      customGoBack={customGoBack}
+      contextualHelp={emptyContextualHelp}
+      headerTitle={I18n.t("features.mvl.details.attachments.pdfPreview.title")}
+    >
+      <SafeAreaView
+        style={styles.container}
+        testID={"FciDocumentPreviewScreenTestID"}
+      >
+        {pipe(fciDownloadPath, S.isEmpty) === false && (
```

Comment:
Is this pipe useful? Isn't S.isEmpty(fciDownloadPath) enough?
Is correct that nothing is rendered when the fciDownloadPath is empty?

Reply:
At the beginning of the render there is a check on `fciDownloadSelector` where `fciDownloadPath` come from. Maybe this check could be unuseful. I'll try to remove it.